frontend/src/scenes/cohorts/cohortEditLogic.ts

```diff
@@ -254,16 +255,46 @@ export const cohortEditLogic = kea<cohortEditLogicType>([
             },
         ],
         duplicatedStaticCohort: [
+            null as CohortType | null,
+            {
+                duplicateToDynamicCohort: async (_, breakpoint) => {
+                    try {
+                        await breakpoint(200)
+                        const data = { ...values.cohort }
+                        data.name += ' (dynamic copy)'
+                        const cohort = await api.cohorts.create(data)
+
+                        lemonToast.success(
+                            'Cohort duplicated dynamically. Please wait up to a few minutes for it to be calculated',
+                            {
+                                toastId: `cohort-dynamically-duplicated-${values.cohort.id}`,
+                                button: {
+                                    label: 'View cohort',
+                                    action: () => {
+                                        router.actions.push(urls.cohort(cohort.id))
+                                    },
+                                },
+                            }
+                        )
+                        return cohort
+                    } catch (error: any) {
+                        lemonToast.error(error.detail || 'Failed to duplicate cohort')
+                        return null
+                    }
+                },
+            },
+        ],
+        duplicatedDynamicCohort: [
             null as CohortType | null,
             {
                 duplicateToStaticCohort: async (_, breakpoint) => {
                     try {
                         await breakpoint(200)
                         const cohort = await api.cohorts.duplicate(values.cohort.id)
                         lemonToast.success(
-                            'Cohort duplicated. Please wait up to a few minutes for it to be calculated',
+                            'Cohort duplicated statically. Please wait up to a few minutes for it to be calculated',
```

Comment:
I don't think you need to add the 'statically' or 'dynamically' qualifiers here or in the toast above.

Reply:
Makes sense, cleaned that up.

Reply:
Also took care of these two items:

> 1. We can combine the two duplicators into one, with a parameter for static/dynamic - since most of the logic is the same.
> 
> 2. The header is getting a bit busy, can you move the "Delete" and "Duplicate as dynamic cohort" buttons into a three-dot menu like we see in experiments / surveys?

I'm not 100% sure I did the new parameter for the combined function right, as I'm still wrapping my head a bit around how it's intended to work. Hopefully I got it right, or close!

Reply:
Also added a few tests. I think it's good for another look whenever you have some time.

Reply:
love it, thanks!